### PR TITLE
JAVA-2158: Allow BuildableQuery to build statement with values

### PR DESCRIPTION
**Pull request by olim7t**
TODO
- [x] add variant for named values
- [x] reword commit message (to match changelog and PR title)

### CODE REVIEW

File: query-builder/src/test/java/com/datastax/oss/driver/api/querybuilder/BuildableQueryTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        deleteFrom("foo").whereColumn("k").isEqualTo(bindMarker()),
+        new Object[] {1},
+        "DELETE FROM foo WHERE k=?",
+        true
+      },
+      {
+        deleteFrom("foo").whereColumn("k").isEqualTo(bindMarker()).ifExists(),
+        new Object[] {1},
+        "DELETE FROM foo WHERE k=? IF EXISTS",
+        false
```
Comment (tomekl007) [inline query-builder/src/test/java/com/datastax/oss/driver/api/querybuilder/BuildableQueryTest.java:60]: why this query is not idempotent? I think that this DELETE should be

File: query-builder/src/test/java/com/datastax/oss/driver/api/querybuilder/BuildableQueryTest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      },
+      {
+        update("foo").setColumn("v", bindMarker()).whereColumn("k").isEqualTo(bindMarker()),
+        new Object[] {3, 1},
+        "UPDATE foo SET v=? WHERE k=?",
+        true
+      },
+      {
+        update("foo")
+            .setColumn("v", function("non_idempotent_func"))
```
Comment (tomekl007) [inline query-builder/src/test/java/com/datastax/oss/driver/api/querybuilder/BuildableQueryTest.java:82]: is there a way to create a function that is `idempotent`?

File: query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/BuildableQuery.java [hunk 3]
```diff
@@ -47,14 +47,32 @@
...
+   *
+   * <pre>{@code
+   * SimpleStatement.newInstance(asCql(), values)
+   * }</pre>
+   *
+   * In addition, some implementations might try to infer additional statement properties (such as
+   * {@link Statement#isIdempotent()}).
+   */
+  @NonNull
+  default SimpleStatement build(Object... values) {
```
Comment (adutra) [inline query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/BuildableQuery.java]: While at it, why not also add variants for named values?

File: query-builder/src/test/java/com/datastax/oss/driver/api/querybuilder/BuildableQueryTest.java (hunk 1 above)
Comment (olim7t) [reply query-builder/src/test/java/com/datastax/oss/driver/api/querybuilder/BuildableQueryTest.java:60]: Conditional queries are always considered non idempotent. This is counter-intuitive, but there are subtle interactions with retries. See explanation [here](http://coppi.datastax.lan/en/developer/java-driver/4.0-rc/manual/query_builder/idempotence/#conditional-statements).

File: query-builder/src/test/java/com/datastax/oss/driver/api/querybuilder/BuildableQueryTest.java (hunk 2 above)
Comment (olim7t) [reply query-builder/src/test/java/com/datastax/oss/driver/api/querybuilder/BuildableQueryTest.java:82]: No. The query builder is pessimistic by default and will always consider the query as idempotent as soon as there is a function call.
If you know that all your functions are safe, the workaround is to override the flag on the statement.

File: query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/BuildableQuery.java (hunk 3 above)
Comment (olim7t) [reply query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/BuildableQuery.java]: Indeed. `SimpleStatement.newInstance` has that variant, so it makes sense to add it here too. Will do.

**Pull request merged**